internal/cluster/kubernetes/namespace_resources_deleter.go

```diff
@@ -0,0 +1,70 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"github.com/banzaicloud/pipeline/pkg/k8sclient"
+	"github.com/goph/emperror"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NamespaceResourcesDeleter struct {
+	logger logrus.FieldLogger
+}
+
+func MakeNamespaceResourcesDeleter(logger logrus.FieldLogger) NamespaceResourcesDeleter {
+	return NamespaceResourcesDeleter{
+		logger: logger,
+	}
+}
+
+func (d NamespaceResourcesDeleter) Delete(k8sConfig []byte, namespace string) error {
+	logger := d.logger.WithField("namespace", namespace)
```

Comment:
We should somehow provide cluster information as well. This might be resolved later, and applies to the other deleters too.

Reply:
Done.